### PR TITLE
move various unused typing-only imports into type-checking blocks

### PR DESCRIPTION
**Pull request by eli-schwartz**

### CODE REVIEW

File: mesonbuild/compilers/compilers.py [hunk 1]
```diff
@@ -24,7 +25,7 @@
 from .. import mesonlib
 from ..mesonlib import (
     HoldableObject,
-    EnvironmentException, MachineChoice, MesonException,
+    EnvironmentException, MesonException,
```
Comment (eli-schwartz) [inline mesonbuild/compilers/compilers.py:28]: *cries*

```
  File "/__w/meson/meson/mesonbuild/compilers/cs.py", line 22, in <module>
    from .compilers import Compiler, MachineChoice, mono_buildtype_args
ImportError: cannot import name 'MachineChoice' from 'mesonbuild.compilers.compilers' (/__w/meson/meson/mesonbuild/compilers/compilers.py)
```

File: mesonbuild/ast/interpreter.py [hunk 2]
```diff
@@ -14,8 +14,11 @@
 
 # This class contains the basic functionality needed to run any interpreter
 # or an interpreter-based tool.
+from __future__ import annotations
+
+import os, sys
```
Comment (dcbaker) [inline mesonbuild/ast/interpreter.py:19]: one import per line, please :)

File: mesonbuild/compilers/compilers.py (hunk 1 above)
Comment (dcbaker) [reply mesonbuild/compilers/compilers.py:28]: _cries too_

**Review by dcbaker — APPROVED** [review]
One small nit, otherwise this looks good to me

File: mesonbuild/ast/interpreter.py (hunk 2 above)
Comment (eli-schwartz) [reply mesonbuild/ast/interpreter.py:19]: :p and here I just moved existing blocks of code around.

**Pull request merged**